### ISSUE TITLE
Allow deletions of messages in the UI

### ISSUE THREAD

**Issue by strangelookingnerd** [camunda/camunda-modeler#4343]
### Problem you would like to solve

Currently it is possible to create `bpmn:message` (as well as escalation, errors...) with via the UI.
For the user there is no way to delete them other than jumping into the XML view.

### Proposed solution

It should be possible to delete messages. Also there should be a warning if the user wants to delete a message that is still referenced.

### Alternatives considered

Delete messages by editing the bmpn XML.

### Additional context

See related issues: #4339 #4341

**Comment by nikku** [comment]
This came up in an internal [a discussion on process duplication](https://camunda.slack.com/archives/C032H77434N/p1745999670597399?thread_ts=1745945015.841799&cid=C032H77434N), resulting in non-unique IDs.

Bottom line: What you cannot see is what you cannot comprehend, you cannot organize, and you cannot fix.

**Comment by nikku** [comment]
Rephrased the issue to focus on all global elements.

**Comment by crobbins215** [comment]
This came up when I copied a process for a slack inbound connector, resulting in the orginal process being triggered https://camunda.slack.com/archives/C02JLRNQQ05/p1758714196796819

Could this be looked into @barmac / @Balazs-PM?

**Comment by barmac** [comment]
Not sure if the feature requested in the top comment solves this problem. In the linked problem, we relied on the generated value feature in the element templates to create a unique identifier for the message. Since the identifier was hidden to the user, they had no chance to notice the duplicate when they copied the diagram. Some solutions that could help in that case are:
1. Make the message name visible and editable in the element template so that user can fix it themself
2. Notify the use of a duplicate in Web Modeler
3. Finally, ensure that message is correlated with the deployed definition and not with other diagrams (so use message name + other properties to correlate)

**Comment by barmac** [comment]
Side note: [The feature request](https://github.com/camunda/camunda-modeler/issues/4343#issue-2337691942) makes sense to me, but I don't think it solves [the problem in the link](https://github.com/camunda/camunda-modeler/issues/4343#issuecomment-3337866422).

**Comment by wollefitz** [comment]
> Notify the use of a duplicate in Web Modeler

As far as I understand, you could end up in the same situation with Desktop Modeler by duplicating a bpmn diagram locally that makes use of an element template referencing a message (for example [Webhook Start Event Connector](https://github.com/camunda/connectors/blob/main/connectors/webhook/element-templates/webhook-connector-start-message.json)), continue working on it and deploying it to the same cluster as the original file. I'd advocate for a solution that spans across the modeling stack and makes the problem users are facing transparent (ie. allowing the management of global definitions in the UI **plus** improvements to the element templates itself via 1 and 3, cc @sbuettner) instead of creating a band-aid solution in Web Modeler. 

This aligns with @nikku's proposal in [this discussion](https://camunda.slack.com/archives/C032H77434N/p1745999670597399?thread_ts=1745945015.841799&cid=C032H77434N):

> **In all modelers**, make it possible to see global elements in the first place, and "manage them" on the process level
> * This is an often asked for feature, and the old Eclipse modeler had it
> * Low coders are even less expected to look into the XML to "organize their messages"
> * Maybe time to prioritize  https://github.com/camunda/camunda-modeler/issues/4343 and a full trophy of related issues that will be fixed by that.

@barmac What do you think?

**Comment by barmac** [comment]
> As far as I understand, you could end up in the same situation with Desktop Modeler by duplicating a bpmn diagram locally that makes use of an element template referencing a message 

That is correct. However, from the Desktop Modeler perspective we can never tell whether we are in a duplicate situation, or the user just continues working on the same diagram. In the Web Modeler, we know that we are dealing with a duplicate when there are several files in the system. That does not contradict your notion that detecting duplicates in the Web Modeler is just a bandaid solution.

**However**, so far within the element templates bindings for root elements we have been actively hiding the messages created for templated elements in the UI so that the user can configure them only via the templated element configuration. They cannot link another message event to a message configured via an element template, and there is a 1:1 relationship between templated message, and templated element. The sole purpose of this system is that one cannot break the templated element via a different element configuration.

So in the implementation of this feature we need to decide that either we keep the system as is, and allow to modify only the non-templated root elements, or we take the risk of breaking the diagram if we allow to edit also templated messages etc. A compromise could be to display the templated messages, but disallow modifying their values.
